testplan/testing/multitest/result.py

```diff
@@ -148,6 +150,13 @@ def wrapper(result, *args, **kwargs):
                                     in assertion_state.line_range
                                 ):
                                     break
+                            else:
+                                raise RuntimeError(
+                                    "Cannot find a matching frame in the call"
+                                    f" stack of function `{func.__name__}`."
+                                    f" Refer to file: {call_stack[1].filename}"
+                                    f" Line: {call_stack[1].lineno}"
+                                )
```

Comment:
This exception is actually getting thrown in the Windows job. Please see if is expected and we should assert it being raised otherwise just fix the test.

Reply:
> News fragment is missing, please add it.

yes, add a news fragment.

Reply:
this exception here is for testing, because I think this for loop sometimes cannot work correctly. At least I found when assertion is written in pre testcase or post testcase, the result is not correct. I have s simple fix for it.